examples/web-wagmi/src/profiles/UseFollowAndUnfollow.tsx

```diff
@@ -9,46 +13,65 @@ import { Loading } from '../components/loading/Loading';
 import { ProfileCard } from './components/ProfileCard';
 
 type FollowButtonProps = {
-  profile: ProfileFragment;
+  follower: ProfileOwnedByMeFragment;
+  followee: ProfileFragment;
 };
 
-function FollowButton({ profile }: FollowButtonProps) {
-  const { follow, isPending: isFollowing } = useFollow({ profile });
-  const { unfollow, isPending: isUnfollowing, error } = useUnfollow({ profile });
+function FollowButton({ followee, follower }: FollowButtonProps) {
+  const {
+    execute: follow,
+    error: followError,
+    isPending: isFollowPending,
+  } = useFollow({ follower, followee });
+  const {
+    execute: unfollow,
+    error: unfollowError,
+    isPending: isUnfollowPending,
+  } = useUnfollow({ follower, followee });
 
-  useEffect(() => {
-    if (error) toast.error(error.message);
-  }, [error]);
+  if (followee.followStatus === null) {
+    return null;
+  }
 
-  if (profile.isFollowedByMe || profile.isOptimisticFollowedByMe)
+  if (followee.followStatus.isFollowedByMe) {
     return (
-      <button onClick={unfollow} disabled={isUnfollowing}>
-        Unfollow
-      </button>
+      <>
+        <button
+          onClick={unfollow}
+          disabled={isUnfollowPending || !followee.followStatus.canUnfollow}
+          title={
+            !followee.followStatus.canUnfollow
+              ? 'The previous follow request is not finalized on-chain just yet.'
+              : undefined
+          }
+        >
+          Unfollow
+        </button>
+        {unfollowError && <p>{unfollowError.message}</p>}
+      </>
     );
+  }
 
   return (
-    <button onClick={follow} disabled={isFollowing}>
-      Follow
-    </button>
-  );
-}
-
-type ProfileFollowProps = {
-  profile: ProfileFragment;
-};
-
-function ProfileFollow({ profile }: ProfileFollowProps) {
-  return (
-    <article>
-      <ProfileCard profile={profile} />
-      <FollowButton profile={profile} />
-    </article>
+    <>
+      <button
+        onClick={follow}
```

Comment:
I remember a discussion on discord to allow to pass profile to follow to the "execute" method, not to define it all as a hook args. is that not relevant anymore?

Reply:
It depends... in general the item that is the subject of the operation has to be passed as hook arg cause the `isPending` will be relevant for the item. This is the case of `followee` in `useFollow`, `publication` in `useCollect` and so on.
Passing the subject as callback arg will need to figure out another way to communicate "pending status for the given item", and why need to do things in 2 ways?!


Some hooks requires a profile as the "actor" performing the operation which is so far always one of the user's profiles, typed as `ProfileOwnedByMeFragment` (I really hate the `Fragment` suffix in those types, but that's for another day).
This is the case of `follower` in `useFollow`, `collect` in `useCollect` (see following PR), etc.

The reason the `profile` is passed as hook arg is to guide the user to make sure they have in scope a profile they own and not just a random profile id. From a pure TS perspective, if you look at the hooks that gives you instances `ProfileOwnedByMeFragment` you are automatically drawn to  use `useActiveProfile` (which is the right thing to do).

The integrator, if they want, can still figure out if they own a random profile from a list via the `isProfileOwnedByMe`, which BTW I just checked and realized I am not re-exporting from `@lens-protocol/react` so going to do it in a follow PR.